tools/mason/MasonNew.chpl

```diff
@@ -91,13 +103,112 @@ proc masonNew(args) throws {
   }
 }
 
+/*
+  Starts an interactive session to create a
+  new library project.
+*/
+proc beginInteractiveSession(isInit: bool, defaultPackageName: string) throws {
```

Comment:
Another way we could do this is:

```chpl
proc beginInteractiveSession(defaultPackageName='') throws {
  ...
}
```

If a package name is passed to the function, we can assume we are using `mason init`. If not, we can assume we are using `mason new`.